include/GafferBindings/SignalBinding.inl

```diff
@@ -352,6 +352,37 @@ struct Slot : public SlotBase<Signal::slot_function_type::arity, Signal, Caller>
 	}
 };
 
+// Ideally we would bind `boost::signals::trackable` to Python
+// directly, but its protected destructor prevents that. So we
+// bind this little derived class instead.
+struct Trackable : public boost::signals::trackable
+{
+};
+
+// Overload boost's `visit_each()` function for all our Slot types.
+// Boost will call this to discover slots which refer to trackable
+// objects, and will use it to automatically remove the connection
+// when the `trackable` object dies.
+template<typename Visitor, typename Signal, typename Caller>
+void visit_each( Visitor &visitor, const Slot<Signal, Caller> &slot, int )
+{
+	// Check to see if slot contains a WeakMethod referring to a trackable
+	// object. There is no point checking for regular methods, because they
+	// prevent the trackable object from dying until it has been disconnected
+	// manually.
+	boost::python::object gaffer = boost::python::import( "Gaffer" );
+	boost::python::object weakMethod = gaffer.attr( "WeakMethod" );
```

Comment:
I'm not sure how often this `visit_each` is called, but it might be worth making these static. @donboie noted a performance improvement for a module import that was being called per-location (https://github.com/ImageEngine/cortex/pull/841/commits/cb8afea895b77b4f7b16f926681e8b451bdae6c0), though I suppose I don't know if the `static` part was actually a factor there...

Reply:
I did wonder about doing this, but I was wary of having a static `boost::python::object`, since it means we'll be trying to decrement a reference to a python object on exit, _after_ python has shut down. That has caused problems before.

It is definitely faster though. Making 100000 connections to a WeakMethod :

Baseline : 0.348s
This PR : 0.542s
Statics : 0.412s

To get an idea of the impact on something in the real world, I just instrumented Gaffer to see how many connections we make when opening a NodeEditor for Arnold's standard_surface - this is the biggest batch of python connection creations I can think of. That makes 4166 connections, so according to the above it would now be`4166 * (0.542 - 0.348) / 100000` seconds slower, which by my reckoning is 0.008s. I think that's OK, but perhaps there's a heavier use case I'm overlooking?

I suspect my objection to the static could be countered by a tactical leak along the lines of `new object( import( "Gaffer" ) )`, but I'd rather keep things simple and worry-free if we can't think of a more performance-critical use case.